docs/notebooks/create_synthetic_data_from_gretel_dgan_api.ipynb

```diff
@@ -0,0 +1,321 @@
+    "for val in list(df.iloc[:,1:].columns):\n",
+    "    plt.figure(figsize = (16,8))\n",
+    "    plt.subplot(1, 2, 1)\n",
+    "    plt.hist([np.array(df[val]), np.array(synthetic_df[val])], \n",
+    "             label=[\"real\", \"synthetic\"],\n",
+    "             bins=50,\n",
+    "             density=True,\n",
+    "             )\n",
+    "    plt.legend()\n",
+    "    plt.xlabel(val)\n",
+    "    plt.ylabel(\"Density\")\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "095ed91d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Functions to calculate autocorrelation which will be visualized below.\n",
+    "def autocorr(X, Y):\n",
+    "    EPS = 1e-8\n",
+    "    Xm = torch.mean(X, 1).unsqueeze(1)\n",
+    "    Ym = torch.mean(Y, 1).unsqueeze(1)\n",
+    "    r_num = torch.sum((X - Xm) * (Y - Ym), 1)\n",
+    "    r_den = torch.sqrt(torch.sum((X - Xm)**2, 1) * torch.sum((Y - Ym)**2, 1))\n",
+    "\n",
+    "    r_num[r_num == 0] = EPS\n",
+    "    r_den[r_den == 0] = EPS\n",
+    "\n",
+    "    r = r_num / r_den\n",
+    "    r[r > 1] = 0\n",
+    "    r[r < -1] = 0\n",
+    "\n",
+    "    return r\n",
+    "    \n",
+    "def get_autocorr(feature):\n",
+    "    feature = torch.from_numpy(feature)\n",
+    "    feature_length = feature.shape[1]\n",
+    "    autocorr_vec = torch.Tensor(feature_length - 2)\n",
+    "\n",
+    "    for j in range(1, feature_length - 1):\n",
+    "      autocorr_vec[j - 1] = torch.mean(autocorr(feature[:, :-j], feature[:, j:]))\n",
+    "\n",
+    "    return autocorr_vec.cpu().detach().numpy()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8c4389c4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#function to generate numpy array in order to visualize the autocorrelation between real and synthetic data\n",
+    "def generate_numpy_for_autocorr(df, batch_size):\n",
+    "    features = df.iloc[: , 1:].to_numpy()\n",
+    "    n = features.shape[0] // batch_size\n",
+    "\n",
+    "    # Shape is now (# examples, # time points, # features)\n",
+    "    features = features[:(n*batch_size),:].reshape(-1, batch_size, features.shape[1])\n",
+    "    return features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9bb282d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Generate autocorrelation features from synthetic and real data and plot!\n",
+    "synthetic_acf = get_autocorr(generate_numpy_for_autocorr(df, seq_len))\n",
+    "acf = get_autocorr(generate_numpy_for_autocorr(synthetic_df, seq_len))\n",
+    "# Figure 1, autocorrelation\n",
+    "plt.plot(acf, label=\"real\")\n",
+    "plt.plot(synthetic_acf, label=\"generated\")\n",
+    "plt.xlabel(\"Time lag (days)\")\n",
+    "plt.ylabel(\"Autocorrelation\")\n",
+    "plt.title(\"Autocorrelation of Heartbeat 1 and Heartbeat 2\")\n",
```

Comment:
Earlier it says the dataset is "oil", but here it says heartbeat 1 and 2, is that right?